client/src/nv_ingest_client/client/client.py

```diff
@@ -425,18 +506,32 @@ def _submit_job(
 
             # Free up memory -- payload should never be used again, and we don't want to keep it around.
             job_state.job_spec.payload = None
-            
+
             return x_trace_id
         except Exception as err:
             logger.error(f"Failed to submit job {job_index} to queue {job_queue_id}: {err}")
             job_state.state = JobStateEnum.FAILED
             raise
 
-    def submit_job(self, job_indices: Union[str, List[str]], job_queue_id: str) -> List[Union[Dict, None]]:
+    def submit_job(
+        self, job_indices: Union[str, List[str]], job_queue_id: str, batch_size: int = 10
+    ) -> List[Union[Dict, None]]:
         if isinstance(job_indices, str):
             job_indices = [job_indices]
 
-        return [self._submit_job(job_id, job_queue_id) for job_id in job_indices]
+        results = []
+        total_batches = math.ceil(len(job_indices) / batch_size)
+
+        for batch_num in range(total_batches):
+            batch_start = batch_num * batch_size
+            batch_end = batch_start + batch_size
+            batch = job_indices[batch_start:batch_end]
+
+            # Submit each batch of jobs
+            batch_results = [self._submit_job(job_id, job_queue_id) for job_id in batch]
```

Comment:
I'm just noticing this now, but we probably need to handle exceptions from _submit_job better. Currently, if we submit a batch of exceptions and one fails, the rest of the items in the batch are failed.

Reply:
Updated in [f2cc046](https://github.com/NVIDIA/nv-ingest/pull/158/commits/f2cc0460f3c80d046db091f19d06a5810cd133ce) and [631d80a](https://github.com/NVIDIA/nv-ingest/pull/158/commits/631d80ad115df8bc8e19333db541590f1aae8122).